### PR TITLE
Model profiling

### PR DESCRIPTION
**Pull request by thesps**
Add utilities for numerical profiling of (`keras`) models.
Produces plots of the distributions of weights of all layer of a model, and optionally activations if provided test data.
Could be used to assist choice of precision.
Introduces a bunch of new imports: pandas, keras, seaborn...

API is like:
```
model = ... # load a model
X = ... # load some data
from hls4ml.model.profiling import numeric
numeric(model, X)
```

e.g. for the Keras 3 layer model with BatchNormalization:
profiling_weights

And activations when provided with some of the jet tagging dataset:
profiling_activation

### CODE REVIEW

**Comment by vloncar** [comment]
This introduces a dependency on pandas, seaborn and matplotlib, which is a bit heavy. I would suggest this gets in as an "extra" to the base package. This can be done by following the instructions [here](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies). Something like this added to `setup()` in `setup.py`:

```
extras_require=[
    'profiling': [
        'pandas',
        'seaborn',
        'matplotlib'
    ]
]
```
Then the optional part could be installed with `pip install hls4ml[profiling]` (or in our case for development as `pip install -e .[profiling]`)

**Comment by thesps** [comment]
Okay cool, nice solution, I've implemented it (additionally requiring keras and tensorflow 🙈).
Tested with a fresh conda environment followed by `pip install -e .[profiling]`, then loading and profiling a model.
